packages/asset-uploader/src/utils/get-asset-path.ts

```diff
@@ -8,18 +8,28 @@ const fsEnv = FsEnv.parse(process.env);
 export const getAssetPath = (asset: DbAsset) => {
   if (asset.location === Location.FS) {
     const splitPath = fsEnv.FILE_UPLOAD_PATH.split("public");
-    return path.join("/", splitPath[splitPath.length - 1], asset.name);
+    const locUrl = new URL(
+      path.join(
+        "/",
+        splitPath[splitPath.length - 1],
+        encodeURIComponent(asset.name)
+      ),
+      // Hostname here is not important
+      "http://localhost"
+    );
+
+    return locUrl.pathname;
   }
 
   if (asset.location === Location.REMOTE && s3Envs.success) {
     if (s3Envs.data.ASSET_CDN_URL) {
       const cndUrl = new URL(s3Envs.data.ASSET_CDN_URL);
-      cndUrl.pathname = encodeURIComponent(asset.name);
+      cndUrl.pathname = encodeURIComponent(encodeURIComponent(asset.name));
```

Comment:
encodeURIComponent twice? Same below.

Reply:
Yep otherwise spaces, cyrillic etc is not working. 
On upload we do following https://github.com/webstudio-is/webstudio-designer/blob/6488305087e661923415971a36d491727e336072/packages/asset-uploader/src/targets/s3/upload.ts#L85 (otherwise s3 will respond with error I think)

Then we need to encode filename into a path, but filename is already `filename = encodeURIComponent(asset.name)` see link above
Having that here we need `encodeUriComponent(filename)` we got 2nd encode ;-)

Reply:
Ok, just wanted to make sure this is not a mistake

Reply:
Ill add my comments into a code as its really confusing and probably need some investigations, like storing fileName on asset